common/lib/capa/capa/tests/test_capa_problem.py

```diff
@@ -25,33 +30,43 @@ def test_label_and_description_inside_responsetype(self):
         xml = """
         <problem>
             <choiceresponse>
-                <label>Select the correct synonym of paranoid?</label>
+                <label>{question}</label>
                 <description>Only the paranoid survive.</description>
                 <checkboxgroup>
                     <choice correct="true">over-suspicious</choice>
                     <choice correct="false">funny</choice>
                 </checkboxgroup>
             </choiceresponse>
         </problem>
-        """
+        """.format(question=question)
         problem = new_loncapa_problem(xml)
         self.assertEqual(
             problem.problem_data,
             {
                 '1_2_1':
                 {
-                    'label': 'Select the correct synonym of paranoid?',
+                    'label': question,
                     'descriptions': {'description_1_1_1': 'Only the paranoid survive.'}
                 }
             }
         )
         self.assertEqual(len(problem.tree.xpath('//label')), 0)
 
-    def test_legacy_problem(self):
+    @ddt.unpack
+    @ddt.data(
+        {
+            'question': 'Once we become predictable, we become ______?',
+            'label_attr': 'Once we become predictable, we become ______?'
+        },
+        {
+            'question': 'Once we become predictable, we become ______?<img src="img/src"/>',
```

Comment:
@cahrens The label is already within response type. In this PR we're picking up the **content** of p tag if that matches with label attribute, earlier we were picking up **text** of p tag. e.g. in master branch the "<p>Question <img></p>" will match with label attribute text "Question", resulting into different positions of "Question" and "<img>". 
Now in this PR content of "p" tag and label attribute won't match since we're comparing content not only text. So we'll pick and set the content of "p" as question label.

Let me know if it's still not clear.